### PR TITLE
Budget-App Capstone

### PR DESCRIPTION
**Pull request by kelomo2502**
In this milestone, I did the followings:

- Adhere to Microverse ROR best practices
- Follow design guidelines as per project requirement
- Implement splash screen interactions
- Sign up and log in pages

-    The user should be able to register in the app with full name, email and password (all mandatory).

-    The user can log into the app using email and password.

-   If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

-       Home page (categories page)

- When the user logs in, they are presented with the categories page.

- For each category, the user can see their name, icon and the total amount of all the transactions that belong to that category.

- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.

- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page

-     For a given category, the list of transactions is presented, ordered by the most recent.

-     At the top of the page, the user could see the total amount for the category  (sum of all of the amounts of the transactions 
      in that category).

-    There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.

-    When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page

-    The user fills out a form to create a new category, indicating their name and icon (both mandatory).

-    The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.

-    When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page

-    The user fills out a form to create a new transaction with:

-    name (mandatory)

-   amount (mandatory)

-   categories (mandatory at least one)

- The user clicks (or taps) the "Save" button to create the transaction, and is taken to the transactions page for that category.

- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

- Add unit test and integration testing to relevant components

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,84 @@
...
+- Languages used
+  - Ruby
+  - HTML
+  - CSS
+  - Rails
+  
+- Technologies used
+  - PostgreSql
+  - Rspec
+  - Capybara
```
Comment (brytebee) [inline README.md:23]: - [ ] **Required:** Great job building the app so far, however, some submission requirements are missing:
    - A recorded video to present the project and mention some code you developed that you are very proud of.
    - A link to the deployed version of your application.

File: README.md [hunk 2]
```diff
@@ -1,24 +1,84 @@
...
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](https://github.com/kelomo2502/budget-app/issues/1).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
```
Comment (brytebee) [inline README.md:80]: -  You should render an "Acknowledgement" or give appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/). It is a professional thing to do when using the resources of other developers in your work.

File: Gemfile [hunk 3]
```diff
@@ -1,46 +1,55 @@
-source "https://rubygems.org"
+source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
```
Comment (brytebee) [inline Gemfile:3]: - [ ] **Optional:** Kindly confirm that the app is working correctly when cloned as I encountered the error below when I attempted to run the code locally.

image

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @kelomo2502👋, 

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- You have used professional repository flow in developing the application.✔️

##

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by EricMbouwe — APPROVED** [review]
Hi @kelomo2502, 

In my opinion, your project is ready for presentation! ⚡🔥 There is nothing else to say other than... it's time to merge it :shipit: 

Good luck! 🎉 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**